packages/xstate-solid/src/createImmutable.ts

```diff
@@ -53,21 +53,23 @@ const updateStore = <Path extends unknown[]>(
       const smallestSize = Math.min(prev.length, next.length);
       const largestSize = Math.max(next.length, prev.length);
 
+      // Update new or now undefined indexes
+      if (newIndices !== 0) {
+        for (let newEnd = smallestSize; newEnd <= largestSize - 1; newEnd++) {
+          set(...path, newEnd, deepClone(next[newEnd]));
```

Comment:
i thought we might only have to deep clone everything once upfront - do we deep clone objects anywhere else while performing other types of *updates* right now?

Reply:
Perhaps my question is - wow is this handled when a new property is being added to an object?

Reply:
From my initial testing, only inserting into a new array index seems to require a deep clone. Moving nested objects around and into new properties works as expected. I will dig deeper into this to see if something is happening with new array indexes that are not happening with object updates. It might only be the root reference that is mutated, which would mean we could just shallow clone, but I want to confirm why the current behavior is happening before considering that.

Reply:
Thanks Austin. I appreciate it ❤️

Reply:
After additional reading/testing, I found that we only needed to place an empty array/object in the new array index; the diff loop properly updates all changes after that. Objects automatically populate with the path provided to set, but arrays require the nested object/array to be set before using the path

Reply:
> Objects automatically populate with the path provided to set, but arrays require the nested object/array to be set before properly updating the path

Why? What is the difference between arrays and objects in this regard? 🤔